src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -1275,6 +1281,21 @@ COOP_PINVOKE_HELPER(Object**, RhGetThreadStaticStorage, ())
     return pCurrentThread->GetThreadStaticStorage();
 }
 
+COOP_PINVOKE_HELPER(Object**, RhGetInlineThreadStaticStorage, ())
+{
+    Thread* pCurrentThread = ThreadStore::RawGetCurrentThread();
+    return pCurrentThread->GetInlineThreadStaticStorage();
+}
+
+COOP_PINVOKE_HELPER(Object*, RhpGetThreadStaticBaseForType, (void* unused, uint32_t typeIndex))
+{
+    Thread* pCurrentThread = ThreadStore::RawGetCurrentThread();
+    Object* pInlineThreadStaticStorage = *pCurrentThread->GetInlineThreadStaticStorage();
+
+    // +2 to skip MT pointer and aligned length
```

Comment:
Is this an array? If so, cast to `Array*` and use `->GetArrayLength()`?

Reply:
this is a sz array. GetArrayData does not know that so it will be more complex than `+2`.

Reply:
We can add GetSzArrayData that asserts the array is an SzArray. And delete GetArrayData because all uses of it are for SzArray.

Bypassing the type system this was feels really unnecessary.

Reply:
This is a helper that was originally implemented in assembly. It may end up being inlined by the JIT and the storage may not end up being an array.  
I think byte math is ok for now - while PR is in progress. I'd not worry about that just yet. 

I doubt that JIT will inline this on all platforms right away though, so we may use this helper in some form for a while.
Depending on what the final version of this helper will be, we can make it more "strongly typed".

Reply:
This is resolved now. The storage is not an array and the helper is not in c++ (since we must guarantee tailcall).